samples/net/lwm2m_client/README.rst

```diff
@@ -6,14 +6,144 @@ LWM2M client
 Overview
 ********
 
-TODO
+Lightweight Machine to Machine (LwM2M) is an application layer protocol
+based on CoAP/UDP. and is designed to expose various resources for reading,
+writing and executing via an LwM2M server in a very lightweight environment.
+
+This LwM2M client sample application for Zephyr implements the LwM2M library
+and establishes both IPv4 and IPv6 connections to a LwM2M server using
+the Open Mobile Alliance "Client Registration Interface".
+
+The source code for this sample application can be found at:
+:file:`samples/net/lwm2m_client`.
+
+Requirements
+************
+
+- :ref:`networking_with_qemu`
+- Linux machine
+- Leshan Demo server (source: https://github.com/eclipse/leshan)
 
 Building and Running
 ********************
 
-TODO
+There are configuration files for various hardware setups in the
+samples/net/lwm2m_client directory:
+
+- :file:`prj_frdm_k64f.conf`
+  Use this for FRDM-K64F board with built-in ethernet.
+
+- :file:`prj_qemu_x86.conf`
+  Use this for x86 QEMU.
+
+Build lwm2m-client sample application like this:
+
+.. code-block:: console
+
+    $ cd $ZEPHYR_BASE/samples/net/lwm2m_client
+    $ make pristine && make CONF_FILE=<your desired conf file> \
+      BOARD=<board to use>
+
+The easiest way to setup this sample application is to build and run it
+via QEMU using the configuration :file:`prj_qemu_x86.conf`.
+This requires a small amount of setup described in :ref:`networking_with_qemu`.
+
+Download and run the latest build of Leshan Demo server:
+
+.. code-block:: console
+
+    $ wget https://hudson.eclipse.org/leshan/job/leshan/lastSuccessfulBuild/artifact/leshan-server-demo.jar
+    $ java -jar ./leshan-server-demo.jar -wp 8080
+
+You can now open a web browser to: http://localhost:8080 This is where you
+can watch and manage connected LwM2M devices.
+
+Build lwm2m-client sample application for QEMU like this:
+
+.. code-block:: console
+
+    $ cd $ZEPHYR_BASE/samples/net/lwm2m_client
+    $ make pristine && make run
+
+The sample will start and automatically connect to Leshan demo server with
+both an IPv4 client endpoint (qemu_x86-ipv4-########) and an IPV6 client
+endpoint (qemu_x86-ipv6-########).  The "########" portion is a randomly
+generated number so you can have multiple clients running at the same time.
 
 Sample output
 =============
 
-TODO
+This is the output from the QEMU console:
+
+.. code-block:: console
+
+    To exit from QEMU enter: 'CTRL+a, x'
+    [QEMU] CPU: qemu32
+    qemu-system-i386: warning: Unknown firmware file in legacy mode:
+    genroms/multiboot.bin
+
+    [lib/lwm2m_engine] [DBG] lwm2m_engine_init: LWM2M engine thread started
+    [lwm2m_obj_security] [DBG] security_create: Create LWM2M security instance:
```

Comment:
This would read better if you join these broken lines (because of line length wrapping in your editor)